language-design.md

```diff
@@ -17,7 +17,78 @@ For each quote, describe what it was about the quote that led you pick it.
+update language features or philosophies, but also because the values and needs of the language's users change and adapt.
+In the same vein, the adoption and continued use of a language can be perceived as a sort of function of the ability for users
+to affect the direction and development of the language. Domain-specific needs change along with technology and society,
+and the language will need to evolve and keep pace with the change,
+or else risk fading to the background and being replaced by another language that--at least for the moment--
+meets users' needs and expectations. When we came into this class, we had our own ideals about programming languages and DSLs,
+in particular that a language if based on a particularly fruitful and foundational philosophy could be 'the right tool' for users;
+but as with most things, designers and developers must continue to communicate with their users
+and gauge the reception of their product in order to continue tuning the language to meet user expectations
+whilst fixing the language within some philosophy or other boundary. This reading has been enlightening about
+_how_ one should go about proposing, creating, and maintaining a DSL.
+
+<!--
+* Languages grow and develop, not only because of the language designers but also because of the users.
+* The adoption of a language can be determined by the ability of its userbase to change the language to its needs
+* I agree with the idea that DSLs or _small-languages_ should be created with flexibility in mind. Domain specific needs change as technology and people change, and the language will either have to keep up or be replaced.
+* When I came into the class, I had ideals about programming languages and DSLs being able to be 'The Right Thing' but this reading has changed my mind about _how_ one should go about proposing and creating a DSL.
+-->
+
+> **Minimize mutability.** Immutable objects are simple, thread-safe, and freely sharable. [Bloch, 2006]
+
+We initially hesitated to agree with this point, but with some debate and discussion between the two of us,
+we realized the benefits and assented. While minimizing mutability might result in the language being
+more verbose and less efficient (in terms of performance), users benefit by having an easier time reasoning about the state
+of objects in the language, which helps to minimize the "conceptual weight," which Bloch also highlights as being important.
+We recognize the importance of performance--after all, gross computation time can make a language virtually unusable--
+but we believe the performance lost from making a language more verbose (at least in the case of minimizing mutability)
+does not outweigh the gains in reducing conceptual weight, which in the long run can reduce time spent
+debugging and reasoning about a program in the language. This quote captures another og Bloch's ideas about separating
```

Comment:
This is another huge benefit - if it takes me 3 months to write the code, another 3 to test it, and then years of support, has that _really_ saved me time from the code that took me 3 weeks to write and test, with minimal support, that takes 3 times as long to run the algorithm?